SetDefaultSinkId/explainer.md

```diff
@@ -0,0 +1,185 @@
+   through the user-agent default device. If nonempty, this ID should be equal to the deviceId attribute of one of the MediaDeviceInfo values
+   returned from enumerateDevices().
+
+Return:
+   A Promise that fulfills with a value of undefined.
+
+Exceptions:
+  NotAllowedError DOMException
+    Returned if a cross-origin sub frame tries to call the API.
+
+  NotFoundError DOMException
+    Returned if the deviceId does not match any audio output device.
+
+  AbortError DOMException
+    Returned if switching the audio output device to the new audio device failed.
+```
+
+This API is accessible from the top-level frame and the same-origin sub frames and allows modification of the default audio output for both
+the top-level frame and all sub frames, regardless of their origins. However, it’s important to note that this change does not affect custom
+audio outputs specified using the setSinkId method in media element or audio context.
+
+Remember to call this API within a secure context (using HTTPS).
+
+### Example
+
+```js
+<!-- index.html -->
+<body>
+  <button id="audioDeviceSelection">
+  <iframe src="presentationLive.html"></iframe>
+  ...
+  <script>
+    const audioContext = new AudioContext();
+    ...
+    let selectedDeviceId = "";
+
+    // App shows audio output devices for a user to select.
+    audioDeviceSelection.addEventListner('click', (list) => {
+      const devices = await navigator.mediaDevices.enumerateDevices();
+      const audioOutputs = devices.filter((device) => device.kind === 'audiooutput');
+
+      // showDialogForDeviceSelection is a dialog that allow user to select
+      // audio device, it could be 'selectAudioOutput' API
+      // (https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/selectAudioOutput)
+      // if the user agent supports it.
+      selected_device = await showDialogForDeviceSelection(audioOutputs);
+      selectedDeviceId = selected_device.deviceId;
+    });
+
+    ...
+
+    // setDefaultSinkId will change the audio output device for the entire frame that includes subframes.
+
+    // If selectedDeviceId is empty string, "", then setDefaultsinkId will revert to using the system default device.
+    await navigator.mediaDevices.setDefaultSinkId(selectedDeviceId);
+
+    // It does not have to call audioContext.setSinkId in order to change audioContext's audio device output.
+  </script>
+</body>
+
+
+<!-- presentationLive.html -->
+<body>
+  ...
+  <iframe src="mediaPlay.html"></iframe>
+</body>
+
+<!-- mediaPlay.html -->
+<body>
+  ...
+  <video id="videoElem" src="https://www.mediaPlayLive.com/123abc"></video>
+  <script>
+    const videoElem = document.getElementById('videoElem');
+
+    // videoElem.sinkId === selectedDeviceId after calling setDefaultSinkId.
+  </script>
+</body>
+```
+
+## Privacy and Security Considerations
+
+### Privacy
+No considerable privacy concerns are expected, but we welcome community feedback.
```

Comment:
Are there any potential side-channel communication concerns? By not changing the exposed default device ID, that helps a lot. Are there any observable side-effects via WebAudio interop with seeing supported bitrates or something similar?

Reply:
I can't think about any side-channel exploit right now, but continue to investigate and open to feedback.

AudioContext has many detailed configuration for audio but when they use the default device, developer don't care about this as default device relies on the system settings and it already can be changed at any time by the user.